Source/Core/Core/IOS/Network/KD/NetKDRequest.h

```diff
@@ -57,19 +60,43 @@ class NetKDRequestDevice : public EmulationDevice
     KD_Download,
     Client,
     Server,
+    CheckMail,
   };
 
+  enum class SchedulerEvent
+  {
+    Mail,
+    Download,
+  };
+
+  NWC24::ErrorCode KDCheckMail(u32* mail_flag_, u32* interval_);
+
   void LogError(ErrorType error_type, s32 error_code);
+  void SchedulerTimer();
+  void SchedulerWorker(SchedulerEvent event);
+
+  static std::string_view GetValueFromCGIResponse(const std::string& response,
+                                                  const std::string& key);
+  static constexpr std::array<u8, 20> MAIL_CHECK_KEY = {0xce, 0x4c, 0xf2, 0x9a, 0x3d, 0x6b, 0xe1,
+                                                        0xc2, 0x61, 0x91, 0x72, 0xb5, 0xcb, 0x29,
+                                                        0x8c, 0x89, 0x72, 0xd4, 0x50, 0xad};
 
-  NWC24::NWC24Config config;
+  NWC24::NWC24Config m_config;
   NWC24::NWC24Dl m_dl_list;
+  NWC24::Mail::WC24SendList m_send_list;
   Common::WorkQueueThread<AsyncTask> m_work_queue;
+  Common::WorkQueueThread<std::function<void()>> m_scheduler_work_queue;
   std::mutex m_async_reply_lock;
   std::mutex m_scheduler_buffer_lock;
   std::queue<AsyncReply> m_async_replies;
   u32 m_error_count = 0;
   std::array<u32, 256> m_scheduler_buffer{};
   // TODO: Maybe move away from Common::HttpRequest?
   Common::HttpRequest m_http{std::chrono::minutes{1}};
+  u32 m_download_span = 2;
```

Comment:
This and `m_mail_span` are accessed on multiple threads.  One thread is writing (the thread `KDCheckMail` is on) and one thread is reading (the scheduler timer).  Should you mark these as atomic?

Reply:
Used the existing `m_scheduler_lock` mutex.